src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * 
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import static org.junit.Assert.assertEquals;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.FindPatientPage;
+import org.openmrs.contrib.qaframework.page.RegistrationEditSectionPage;
+import org.openmrs.contrib.qaframework.page.RegistrationSummaryPage;
+
+public class EditDemographicsAndContactInfoSteps extends Steps {
+
+	private TestData.PatientInfo testPatient;
+	private RegistrationEditSectionPage registrationEditSectionPage;
+	private RegistrationSummaryPage registrationSummaryPage;
+
+	@Before(RunTest.HOOK.SELENIUM_PATIENT_DEMOGRAPHICS)
+	public void visitPatientDashboard() {
+		testPatient = createTestPatient();
+		initiateWithLogin();
+		findPatientPage = (FindPatientPage) homePage.goToFindPatientRecord().waitForPage();
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_PATIENT_DEMOGRAPHICS)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Edit Registration Information link from Patient dashboard")
+	public void loadRegistrationSummaryPage() {
+		registrationSummaryPage = (RegistrationSummaryPage) dashboardPage.goToRegistrationSummary().waitForPage();
+	}
+
+	@Then("the system loads Registration Summary Page")
+	public void systemLoadsRegistrationSummaryPage() {
+		assertPage(registrationSummaryPage.waitForPage());
+	}
+
+	@When("a user clicks on Edit link from Registration Summary Page")
+	public void loadEditDemographicsSection() {
+		registrationEditSectionPage = (RegistrationEditSectionPage) registrationSummaryPage.clickOnEditDemographics().waitForPage();
+	}
+
+	@And("the system loads edit demographics section")
+	public void systemLoadsEditDemographicsSection() {
+		assertPage(registrationEditSectionPage.waitForPage());
+	}
+
+	@And("a user edits existing demographics")
+	public void editDemographics() {
+		testPatient.givenName = "Jane";
+		testPatient.middleName = "Adyeri";
+		testPatient.familyName = "Namata";
+		testPatient.gender = "Male";
```

Comment:
is `Jane Adyeri Namata` a male patient?

Reply:
@kdaud  that's the patient's preferred gender. Which I think does not break the test.

Reply:
> that's the patient's preferred gender. Which I think does not break the test.

yes it doesn't break the test! However, isn't  `Jane Adyeri Namata` supposed to be a female person?